toml/_parser.ts

```diff
@@ -134,20 +134,13 @@ function failure(): Failure {
   return { ok: false };
 }
 
-export function unflat(
+export function unflat<T extends unknown[] | Record<string, unknown>>(
   keys: string[],
-  values: unknown = {},
-  cObj?: unknown,
-): Record<string, unknown> {
-  const out: Record<string, unknown> = {};
-  if (keys.length === 0) {
-    return cObj as Record<string, unknown>;
-  }
-  if (!cObj) cObj = values;
-  const key: string | undefined = keys[keys.length - 1];
-  if (typeof key === "string") out[key] = cObj;
-  return unflat(keys.slice(0, -1), values, out);
+  values: T,
+): T {
```

Comment:
I think the return type should be `Record<string, unknown>` instead of `T extends unknown[] | Record<string, unknown>`. Updating